File: Data Structures & Algorithms/insert-into-a-binary-search-tree/submission-1.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

// func insertIntoBST(root *TreeNode, val int) *TreeNode {
//     if root == nil {
// 		return &TreeNode{
// 			Val: val,
// 		}
// 	}

// 	if val > root.Val {
// 		root.Right = insertIntoBST(root.Right, val)
// 	} else if val < root.Val {
// 		root.Left = insertIntoBST(root.Left, val)
// 	}
// 	return root
// }

func insertIntoBST(root *TreeNode, val int) *TreeNode {
	newNode := &TreeNode{
		Val: val,
	}

	if root == nil {
		return newNode
	}

	cur := root
	for {
		if val > cur.Val {
			if cur.Right == nil {
				cur.Right = newNode
				return root
			}
			cur = cur.Right
		} else {
			if cur.Left == nil {
				cur.Left = newNode
				return root
			}
			cur = cur.Left
		}
	}


}

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        count = 0

        def dfs(node, max_prev):
            nonlocal count
            if not node:
                return

            if node.val >= max_prev:
                count+=1

            dfs(node.left, max(node.val, max_prev))
            dfs(node.right, max(node.val, max_prev))
            
        dfs(root, root.val)
        return count
        